process/sync/storageBootstrap/shardStorageBootstrapper.go

```diff
@@ -143,12 +143,20 @@ func (ssb *shardStorageBootstrapper) cleanupNotarizedStorage(shardHeaderHash []b
 	}
 }
 
-func (ssb *shardStorageBootstrapper) applySelfNotarizedHeaders(selfNotarizedHeadersHashes [][]byte) ([]data.HeaderHandler, error) {
-	selfNotarizedHeaders := make([]data.HeaderHandler, 0, len(selfNotarizedHeadersHashes))
+func (ssb *shardStorageBootstrapper) applySelfNotarizedHeaders(
+	bootstrapHeadersInfo []bootstrapStorage.BootstrapHeaderInfo,
+) ([]data.HeaderHandler, [][]byte, error) {
+
+	selfNotarizedHeadersHashes := make([][]byte, len(bootstrapHeadersInfo))
+	for index, selfNotarizedHeader := range bootstrapHeadersInfo {
+		selfNotarizedHeadersHashes[index] = selfNotarizedHeader.Hash
+	}
+
+	selfNotarizedHeaders := make([]data.HeaderHandler, len(selfNotarizedHeadersHashes))
 	for _, selfNotarizedHeaderHash := range selfNotarizedHeadersHashes {
 		selfNotarizedHeader, err := ssb.getHeader(selfNotarizedHeaderHash)
 		if err != nil {
-			return nil, err
+			return nil, nil, err
 		}
 
 		selfNotarizedHeaders = append(selfNotarizedHeaders, selfNotarizedHeader)
```

Comment:
this append does not work as expected. The length of the returned selfNotarizedHeaders will be 2 * len(selfNotarizedHeaderHashes). Instead of append you can use selfNotarizedHeaders[index] = selfNotarizedHeader

Reply:
Yep, good catch. Refactored